### PR TITLE
Implement pinning in CLI

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -51,6 +52,8 @@ extern crate ethabi_contract;
 #[macro_use]
 extern crate ethabi_derive;
 
+//extern crate jsonrpc_core as jsonrpc;
```
Comment (DieMyst) [inline cli/src/main.rs]: delete this?

File: cli/src/publisher.rs [hunk 2]
```diff
@@ -126,37 +129,54 @@ impl Publisher {
     }
 }
 
+//fn validate_args(matches: &ArgMatches) -> Result<SomeDTO, AppError> {
```
Comment (DieMyst) [inline cli/src/publisher.rs]: delete this?

File: cli/src/publisher.rs [hunk 3]
```diff
@@ -126,37 +129,54 @@ impl Publisher {
...
+//
+//    let archive = matches.value_of(ARCHIVE_PATH).unwrap().to_string();
+//    let mut archive = File::open(archive).map_err(|e| format!("can't open archive file, check if it exists: {}", e))?;
+//    ...
+//}
+
 /// Creates `Publisher` from arguments
 pub fn parse(matches: &ArgMatches) -> Result<Publisher, Box<Error>> {
-    let path = matches.value_of(PATH).unwrap().to_string();
+    let path = matches.value_of(PATH).unwrap().to_string(); //TODO use is_file from clap_validators
```
Comment (DieMyst) [inline cli/src/publisher.rs]: is it possible to use `?` instead of `unwrap()`?

File: cli/src/publisher.rs [hunk 4]
```diff
@@ -395,4 +420,25 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn publish_pinned_to_unknown_nodes() -> () {
```
Comment (DieMyst) [inline cli/src/publisher.rs]: success test for pinning?

**Comment by folex** [comment]
I'll close this PR and will create a new one, with app deletion & pinning together.

**Pull request closed without merging**